make/hotspot/lib/CompileJvm.gmk

```diff
@@ -156,7 +152,36 @@ $(eval $(call SetupJdkLibrary, BUILD_LIBJVM, \
     abstract_vm_version.cpp_CXXFLAGS := $(CFLAGS_VM_VERSION), \
     arguments.cpp_CXXFLAGS := $(CFLAGS_VM_VERSION), \
     DISABLED_WARNINGS_gcc := $(DISABLED_WARNINGS_gcc), \
+    DISABLED_WARNINGS_gcc_assembler_aarch64.cpp := misleading-indentation, \
+    DISABLED_WARNINGS_gcc_c1_LIR.cpp := misleading-indentation, \
+    DISABLED_WARNINGS_gcc_dict.cpp := char-subscripts, \
+    DISABLED_WARNINGS_gcc_javaClasses.cpp := misleading-indentation, \
+    DISABLED_WARNINGS_gcc_loopnode.cpp := sequence-point, \
+    DISABLED_WARNINGS_gcc_postaloc.cpp := address, \
+    DISABLED_WARNINGS_gcc_sharedRuntimeTrig.cpp := misleading-indentation, \
+    DISABLED_WARNINGS_gcc_shenandoahBarrierSetAssembler_aarch64.cpp := misleading-indentation, \
```

Comment:
```suggestion
    DISABLED_WARNINGS_gcc_shenandoahBarrierSet.cpp := misleading-indentation, \
    DISABLED_WARNINGS_gcc_shenandoahBarrierSetAssembler_aarch64.cpp := misleading-indentation, \
```

I got the following warning for `client` build, i.e. `--disable-precompiled-headers --with-debug-level=release --with-jvm-variants=client`.
```
=== Output from failing command(s) repeated here ===
* For target hotspot_variant-client_libjvm_objs_shenandoahBarrierSet.o:
In file included from /tmp/jdk-test/jdk_src/src/hotspot/share/gc/shenandoah/henandoahBarrierSet.cpp:34:
/tmp/jdk-test/jdk_src/src/hotspot/share/gc/shenandoah/1/shenandoahBarrierSetC1.hpp: In member function 'virtual void LIR_OpShenandoahCompareAndSwap::visit(
/tmp/jdk-test/jdk_src/src/hotspot/share/gc/shenandoah/1/shenandoahBarrierSetC1.hpp:162:7: error: this 'if' clause does not guard... [-Werror=misleading-indentation]
  162 |       if (_info)                    state->do_info(_info);
      |       ^~
/tmp/jdk-test/jdk_src/src/hotspot/share/gc/shenandoah/1/shenandoahBarrierSetC1.hpp:163:37: note: ...this statement, but the latter is misleadingly indented as if 
  163 |                                     state->do_input(_addr);
      |                                     ^~~~~
cc1plus: all warnings being treated as errors

* All command lines available in /tmp/jdk-test/build-release/make-support/failure-logs.
=== End of repeated output ===
```

Reply:
Oh, client builds! I forgot about those. Thanks for reminding me. I'll test the full set of JVM variants as well.